AUTH48

bifurcation opened the pull request:
This PR contains the RFC Editor's markdown as an initial commit.  The author team will review this PR as with any other changes, and make any responses to the RFC Editor as comments or further commits on this PR.

Once this PR is merged, we will indicate to the RFC Editor that the Markdown is approved for publication.

bifurcation commented on draft-ietf-sframe-enc.md:
```diff
@@ -1010,83 +1112,97 @@ approaches to mitigating this risk:
 * Reference: The document where this cipher suite is defined
 
+* Change Controller: Who is authorized to update the row in the registry
+
+<!--[rfced] IANA Considerations. FYI, we have added "Change Controller" to the registry template and table as IANA has added this column to the "SFrame Cipher Suites" registry. Please review. -->
+
```
Thanks, yes, we agreed to this with IANA on that thread.

```suggestion
```

bifurcation commented on draft-ietf-sframe-enc.md:
```diff
@@ -1010,83 +1112,97 @@ approaches to mitigating this risk:
 encryption and decryption operations, and how SFrame ciphertexts are delivered
 from sender to receiver (including any fragmentation and reassembly).  In this
-section, we lay out additional requirements that an integration must meet in
+section, we lay out additional requirements that an implementation must meet in
```
```suggestion
section, we lay out additional requirements that an application must meet in
```

bifurcation commented on draft-ietf-sframe-enc.md:
```diff
@@ -1150,12 +1266,18 @@ As mentioned in {{replay}}, senders MUST reject requests to encrypt multiple tim
 
 ## Metadata
 
+<!--[rfced] Section 9.4. It is unclear what "pure" means in the following sentence. Is the metadata only specified by the application? Please help us clarify.
+
+Original:
+   The metadata input to SFrame operations is pure application-specified
+   data. -->
+
```
Yes, it is whatever the application provides, not modified or processed by the SFrame logic except to feed it into the AAD.  Proposed rewording.

```suggestion
```

bifurcation commented on draft-ietf-sframe-enc.md:
```diff
@@ -1150,12 +1266,18 @@ As mentioned in {{replay}}, senders MUST reject requests to encrypt multiple tim
+   The metadata input to SFrame operations is pure application-specified
+   data. -->
+
 The `metadata` input to SFrame operations is pure application-specified data. As
```
```suggestion
The `metadata` input to SFrame operations comprises arbitrary data specified by the application. As
```

bifurcation commented on draft-ietf-sframe-enc.md:
```diff
@@ -1168,20 +1290,19 @@ processing pipeline, but only to prepare for SFrame decryption.
-The authors wish to specially thank Dr. Alex Gouaillard as one of the early
-contributors to the document. His passion and energy were key to the design and
-development of SFrame.
+<!--Informative References:
+[I-D.codec-agnostic-rtp-payload-format] replaced by draft-gouaillard-avtcore-codec-agn-rtp-payload, which is expired.
+[I-D.ietf-moq-transport] Active WG document
+[I-D.ietf-webtrans-overview] Active WG document
+[MLS-ARCH] Active WG document -->
```
TODO(RLB): Remove this comment.

bifurcation commented on draft-ietf-sframe-enc.md:
```diff
@@ -1201,16 +1322,16 @@ context to be used for sending also tracks the next CTR value to be used.
-  used to initialize a send context, together with a zero counter value.
-* **Adding a key for receiving:** The key and salt are derived from the base key, and
-  used to initialize a send context.
+* **Add a key for sending:** The key and salt are derived from the base key, and
+  are used to initialize a send context, together with a zero counter value.
+* **Add a key for receiving:** The key and salt are derived from the base key, and
+  are used to initialize a send context.
```
```suggestion
* **Add a key for sending:** The key and salt are derived from the base key and
  used to initialize a send context, together with a zero counter value.
* **Add a key for receiving:** The key and salt are derived from the base key and
  used to initialize a send context.
```

bifurcation commented on draft-ietf-sframe-enc.md:
```diff
@@ -1270,27 +1391,34 @@ Here the constant value `1` reflects the fixed SFrame header; `|CTR|` and
-bound on those seen in practice.
+overhead so that the overhead amounts we compute here are likely to be an upper
+bound of those seen in practice.
+
+<!--[rfced] Section B.1. FYI, we added a title to Table 3.
+
+Current:
+  Table 3:  Overhead Analysis Assumptions
+-->
```
👍 

```suggestion
```

bifurcation commented on draft-ietf-sframe-enc.md:
```diff
@@ -1270,27 +1391,34 @@ Here the constant value `1` reflects the fixed SFrame header; `|CTR|` and
 | Key ID (KID)    | 2     | >255 senders; or MLS epoch (E=4) and >16 senders  |
 | Counter (CTR)   | 3     | More than 24 hours of media in common cases       |
-| Cipher overhead | 16    | Full GCM tag (longest defined here)               |
+| Cipher overhead | 16    | Full Galois/Counter Mode (GCM) tag (longest defined here)  |
```
```suggestion
| Cipher overhead | 16    | Full authentication tag (longest defined here)  |
```

bifurcation commented on draft-ietf-sframe-enc.md:
```diff
@@ -1299,65 +1427,65 @@ In audio streams, there is typically a one-to-one relationship between frames
-| FB speech, 20ms packets  |  50 |        32 |           8.6 |      26.9% |
-| FB stereo, 10ms packets  | 100 |       128 |          17.2 |      13.4% |
-{: #audio-overhead title="SFrame overhead for audio streams" }
+| Scenario                  | Frames per Second (fps) | Base kbps | Overhead kbps | Overhead % |
+|:--------------------------|:---:|:---------:|:-------------:|:----------:|
+| NB speech, 120 ms packets | 8.3 |         8 |           1.4 |      17.9% |
+| FB speech, 20 ms packets  |  50 |        32 |           8.6 |      26.9% |
+| FB stereo, 10 ms packets  | 100 |       128 |          17.2 |      13.4% |
+{: #audio-overhead title="SFrame Overhead for Audio Streams" }
```
TODO(RLB): Add "fps" terminology above, revert expansion, and realign column dividers.

bifurcation commented on draft-ietf-sframe-enc.md:
```diff
@@ -1299,65 +1427,65 @@ In audio streams, there is typically a one-to-one relationship between frames
-modern video codecs across a range from very low to very high quality.
+applied per frame and per packet, respectively.  The choices of resolution,
+frames per second, and bandwidth roughly reflect the capabilities of
+modern video codecs across a range from very-low to very-high quality.
```
```suggestion
modern video codecs across a range from very low to very high quality.
```

bifurcation commented on draft-ietf-sframe-enc.md:
```diff
@@ -1299,65 +1427,65 @@ In audio streams, there is typically a one-to-one relationship between frames
 
 Real conferences usually involve several audio and video streams.  The overhead
-of SFrame in such a conference is the aggregate of the overhead over all the
+of SFrame in such a conference is the aggregate of the overhead of all the
```
```suggestion
of SFrame in such a conference is the aggregate of the overhead across all the
```

bifurcation commented on draft-ietf-sframe-enc.md:
```diff
@@ -1378,11 +1506,11 @@ integration between SFrame and RTP could be done, and some of the challenges
+In RTP-based applications, applying SFrame per packet means that the payload of
+each RTP packet will be an SFrame ciphertext, starting with an SFrame header, as
+shown in {{sframe-packet}}.  Applying SFrame per frame means that different
+RTP payloads will have different formats: the first payload of a frame will
```
```suggestion
RTP payloads will have different formats: The first payload of a frame will
```

bifurcation commented on draft-ietf-sframe-enc.md:
```diff
@@ -1393,11 +1521,19 @@ a mechanism for distributing this configuration information. In applications
 ciphertext (except for the SFrame header).  In order for such a generic
-packetization scheme to work interoperably one would have to be defined, e.g.,
+packetization scheme to work interoperably, one would have to be defined, e.g.,
 as proposed in {{?I-D.codec-agnostic-rtp-payload-format}}.
```
```suggestion
as proposed in {{?I-D.gouaillard-avtcore-codec-agn-rtp-payload}}.
```

bifurcation commented on draft-ietf-sframe-enc.md:
```diff
@@ -1393,11 +1521,19 @@ a mechanism for distributing this configuration information. In applications
 extension to SDP could provide this function.
 
-Applying SFrame per-frame also requires that packetization and depacketization
+<!--[rfced] Section B.5. draft-codec-agnostic-rtp-payload-format has been replaced by draft-gouaillard-avtcore-codec-agn-rtp-payload (which expired Sept 2021). Do you want to replace the reference?
+
+Original:
+   In order for such a generic packetization scheme to work
+   interoperably one would have to be defined, e.g., as proposed in
+   [I-D.codec-agnostic-rtp-payload-format].
+-->
+
```
Yep.

```suggestion
```

bifurcation commented on draft-ietf-sframe-enc.md:
```diff
@@ -1393,11 +1521,19 @@ a mechanism for distributing this configuration information. In applications
+Applying SFrame per frame also requires that packetization and depacketization
 be done in a generic manner that does not depend on the media content of the
-packets, since the content being packetized / depacketized will be opaque
+packets, since the content being packetized/depacketized will be opaque
```
```suggestion
packets, since the content being packetized or depacketized will be opaque
```

bifurcation commented on draft-ietf-sframe-enc.md:
```diff
@@ -1557,3 +1693,85 @@ The other values in the test vector are intermediate values provided to
+contributors to the document. His passion and energy were key to the design and
+development of SFrame.
+
+<!--[rfced] Terminology.
```
TODO(RLB): Remove this comment.

bifurcation commented on draft-ietf-sframe-enc.md:
```diff
@@ -908,59 +1010,59 @@ protect against such attacks.
+frames (one per simulcast layer), which would be processed independently by the
 frame encryptor and assigned an unique counter for each.
 
 ### SVC
```
```suggestion
### Scalable Video Coding (SVC)
```

bifurcation commented on draft-ietf-sframe-enc.md:
```diff
@@ -1557,3 +1693,85 @@ The other values in the test vector are intermediate values provided to
+
+b) Please help us expand or define the following term:
+
+   SVC
```
"Scalable Video Coding".  I fixed this above.

bifurcation commented on draft-ietf-sframe-enc.md:
```diff
@@ -1557,3 +1693,85 @@ The other values in the test vector are intermediate values provided to
+
+   SVC
+
+c) May we make the use of the acronym KID consistent?
+
+   Key ID (13) / KID (85)
+
+d) Please help us make the capitalization of the following terms consistent
+
+   counter (22) / Counter (9) - We note that Key ID is capitalized.
+                                Should CTR be used instead?
+
+e) Please review the use of fixed-width format for the following terms and let us know if any updates are necessary:
+
+   context, when referring to a value
+   CTR
+   key, when referring to input
+   KID
+   nonce
```
TODO(RLB): Review these for consistency.

bifurcation replied:
* CTR and KID should never be fixed-width.  It does not appear they ever are, outside of pseudocode.
* "context" should only be fixed width in the context of the MLS section, where it refers to a specific value.
* "key" and "nonce" look to be correctly only fixed-width in the context of the test vectors.

bifurcation commented on draft-ietf-sframe-enc.md:
```diff
@@ -1557,3 +1693,85 @@ The other values in the test vector are intermediate values provided to
+   KID
+   nonce
+
+f) As "up to" may be difficult for ESL readers to understand, may we update the use of the phrase "it is up to the application" to "it is the application's responsibility"?
+
```
TODO(RLB): Implement this suggestion.

bifurcation reviewed (commented):
This is my initial review of the RFC Editor's changes.  Mostly minor stuff, a few rewordings, no major disagreements.  I have made `TODO(RLB)` notations for things that couldn't be handled in GitHub suggestions, and will come back to implement those shortly.

bifurcation commented on draft-ietf-sframe-enc.md:
```diff
@@ -1557,3 +1585,85 @@ The other values in the test vector are intermediate values provided to
+
+c) May we make the use of the acronym KID consistent?
+
+   Key ID (13) / KID (85)
```
Generally, Key ID should be used throughout the section defining the header format, and KID elsewhere.  I have tried to implement this change.

bifurcation commented on draft-ietf-sframe-enc.md:
```diff
@@ -1557,3 +1585,85 @@ The other values in the test vector are intermediate values provided to
+d) Please help us make the capitalization of the following terms consistent
+
+   counter (22) / Counter (9) - We note that Key ID is capitalized.
+                                Should CTR be used instead?
```
When referring to the SFrame field, "Counter" should be used in the section defining the header format, and "CTR" elsewhere.  There are also several generic uses of "counter", which should be left as "counter".

martinthomson commented on draft-ietf-sframe-enc.md:
```diff
@@ -34,8 +31,8 @@ author:
  -
-    ins: R.L. Barnes
-    name: Richard L. Barnes
+    ins: R. Barnes
```
I generally avoid "ins", unless xml2rfc is going to do the wrong thing with "name".  It won't in this case.

```suggestion
```

martinthomson commented on draft-ietf-sframe-enc.md:
```diff
@@ -1299,65 +1285,65 @@ In audio streams, there is typically a one-to-one relationship between frames
-* Narrow-band speech: 120ms packets, 8Kbps
-* Full-band speech: 20ms packets, 32Kbps
-* Full-band stereo music: 10ms packets, 128Kbps
+* Narrow-band (NB) speech: 120 ms packets (8 frames per second (fps)), 8 kbps
```
```suggestion
* Narrow-band (NB) speech: 120 ms packets (8.3 frames per second (fps)), 8 kbps
```

? Why repeat this when it is in the table?

bifurcation replied:
Yeah, just to do the fps expansion outside the table.  I have reworked this table to remove the bullets entirely and put the "fps" expansion in the intro text.

martinthomson reviewed (commented):
Markdown edits FTW.

ajeanmahoney commented on draft-ietf-sframe-enc.md:
```diff
@@ -541,11 +581,11 @@ Header |   | KID |  |                     |
 collects all the fragments of the ciphertext, using appropriate sequencing
 and start/end markers in the transport. Once all of the required fragments are
-available, the client reassembles them into the SFrame ciphertext, then passes
+available, the client reassembles them into the SFrame ciphertext, then it passes
```
Yes, this edit works also.

ajeanmahoney commented on draft-ietf-sframe-enc.md:
```diff
@@ -801,18 +861,30 @@ group.
+effectively defines a reordering window, since no more than 2<sup>`E`</sup>
 epochs can be active at a given time.  Receivers MUST be prepared for the epoch
 counter to roll over, removing an old epoch when a new epoch with the same E
 lower bits is introduced.
```
Perhaps "To handle the rollover of the epoch counter..."

bifurcation commented:
Latest commit addresses comments from @martinthomson, including reworking the audio frame table, removing `inc:` entries from authors and contributors and adding proper accents for @FredericJacobs.

bifurcation commented on draft-ietf-sframe-enc.md:
```diff
@@ -723,17 +783,17 @@ framework, as described in {{key-management-framework}}.
 ## Sender Keys
 
-If the participants in a call have a pre-existing E2E-secure channel, they can
+If the participants in a call have a preexisting E2E-secure channel, they can
```
Well OK, if Chicago and M-W say so...

juberti commented on draft-ietf-sframe-enc.md:
```diff
@@ -159,11 +142,12 @@ media in a broad range of scenarios, as outlined by the following goals:
-4. Independence from the underlying transport, including use in non-RTP
-   transports, e.g., WebTransport {{?I-D.ietf-webtrans-overview}}.
+4. Decouple the media encryption framework from the underlying transport,
+   allowing use in non-RTP, e.g., WebTransport
```
```suggestion
   allowing use in non-RTP transports, e.g., WebTransport
```

juberti commented on draft-ietf-sframe-enc.md:
```diff
@@ -391,7 +414,7 @@ aspects of the AEAD and the hash algorithm below:
   size of a "tag" that is added to the plaintext)
 
 * `AEAD.Nka` - For cipher suites using the compound AEAD described in
```
I originally read `Nka` as "size-of-key-for-AEAD", which made sense to me, but then I realized that since we're always using AEAD, it's really "size-of-key-for-compound-AEAD".  Maybe `Nkc`?

youennf replied:
`Nke` looks good to me.

juberti commented on draft-ietf-sframe-enc.md:
```diff
@@ -908,52 +894,52 @@ protect against such attacks.
 dependencies among frames, e.g., audio streams.  In these streams, each frame is
-independently decodeable, so there is never a need to process two frames
-together which might be on two sides of a key rotation.
+independently decodable, so there is never a need to process together two frames
```
FWIW, I preferred the previous text, or perhaps "process two frames together that are on either side of a key rotation".

bifurcation replied:
Thanks, that's a better rewording.

juberti commented on draft-ietf-sframe-enc.md:
```diff
@@ -1150,14 +1131,14 @@ As mentioned in {{replay}}, senders MUST reject requests to encrypt multiple tim
 
-The `metadata` input to SFrame operations is pure application-specified data. As
-such, it is up to the application to define what information should go in the
+The `metadata` input to SFrame operations an opaque byte string specified by the application. As
```
```suggestion
The `metadata` input to SFrame operations is an opaque byte string specified by the application. As
```

juberti reviewed (approved):
Read over all the changes and overall the edits all look great, the text is clear and easy to follow. Noted a couple minor nits.

murillo128 reviewed (approved):
lgtm, although i think that we are still inconsistently using `Key Id` an `KID` and `Counter` and `CTR`. We also use `Counter (CTR)` in several places.

I think we should either introduce the terms once and then use the abbreviations everywhere, or not use the abbreviations at all.

juberti commented:
@eomara did you mean to merge this? There are still outstanding discussions.

bifurcation commented:
Agree that this should not have been merged quite yet.  I'll start up a new PR to handle the remaining comments.
